cirq-superstaq/cirq_superstaq/service.py

```diff
@@ -961,6 +964,12 @@ def submit_aces(
         elif isinstance(noise, cirq.NoiseModel):
             noise_dict["cirq_noise_model"] = cirq.to_json(noise)
 
+        if weights is not None:
```

Comment:
@richrines1  Where did you suggest we should move this validation logic to?

Reply:
if the weights definitely have to be positive integers we could use something like:
```python
if weights is not None:
    weights = list(weights)
    for weight in weights:
        gss.validation.validate_integer_param(weight, min_val=1)
```
(imo it's probably not necessary to explicitly check that weights is Iterable - it should be obvious enough from the type annotation, and if not `list(weights)` will throw a pretty clear error if given a non-iterable anyway)

Reply:
@natibek  As @richrines1 said IRL, have the validator handle this.